Essentials/src/com/earth2me/essentials/commands/Commandpaytoggle.java

```diff
@@ -15,17 +15,48 @@ public Commandpaytoggle() {
 
     @Override
     public void run(Server server, User user, String commandLabel, String[] args) throws Exception {
-        boolean acceptingPay = !user.isAcceptingPay();
-        if (commandLabel.contains("payon")) {
-            acceptingPay = true;
-        } else if (commandLabel.contains("payoff")) {
-            acceptingPay = false;
-        }
-        user.setAcceptingPay(acceptingPay);
-        if (acceptingPay) {
-            user.sendMessage(tl("payToggleOn"));
+        if (args.length < 1) {
+            boolean acceptingPay = !user.isAcceptingPay();
+            if (commandLabel.contains("payon")) {
+                acceptingPay = true;
+            } else if (commandLabel.contains("payoff")) {
+                acceptingPay = false;
+            }
+            user.setAcceptingPay(acceptingPay);
+            if (acceptingPay) {
+                user.sendMessage(tl("payToggleOn"));
+            } else {
+                user.sendMessage(tl("payToggleOff"));
+            }
         } else {
-            user.sendMessage(tl("payToggleOff"));
+            if (args[0].equalsIgnoreCase("players")) {
+                StringBuilder sb = new StringBuilder();
+                for (String s : user._getBlockingIndividualPay()) {
+                    sb.append(s).append(" ");
+                }
+                String payBlockedList = sb.toString().trim();
+                user.sendMessage(payBlockedList.length() > 0 ? tl("payBlockedList", payBlockedList) : tl(
+                        "noPayBlocked"));
+            } else {
+                User player;
+                try {
+                    player = getPlayer(server, args, 0, true, true);
+                } catch (PlayerNotFoundException ex) {
+                    player = ess.getOfflineUser(args[0]);
+                }
+                if (player == null) {
+                    throw new PlayerNotFoundException();
```

Comment:
Don't catch and re-throw this. If getPlayer doesn't find it, just let it throw itself.